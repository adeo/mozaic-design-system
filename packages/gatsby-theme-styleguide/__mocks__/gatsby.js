const React = require(`react`)
const gatsby = jest.requireActual(`gatsby`)

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
}
