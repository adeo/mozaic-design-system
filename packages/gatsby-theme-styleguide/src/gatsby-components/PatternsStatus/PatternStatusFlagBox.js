import React from 'react'

export default ({ value }) => {
  const flagTypeClass =
    value === 'stable'
      ? 'patternstatusflags__flag-stable'
      : 'patternstatusflags__flag-beta'

  return (
    <span className={'patternstatusflags__flag ' + flagTypeClass}>{value}</span>
  )
}
