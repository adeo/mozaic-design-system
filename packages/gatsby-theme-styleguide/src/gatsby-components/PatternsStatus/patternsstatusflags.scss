.patternstatusflags {
  width: 100%;
  display: flex;
  flex-direction: row;

  &__item {
    margin: 0;
  }

  &__item + &__item {
    margin-left: 50px;
  }
}

.patternstatusflags__flag {
  border-radius: 3px;
  color: #fff;
  display: inline-block;
  margin-left: 8px;
  padding: 2px 7px;
}

.patternstatusflags__flag-beta {
  background: #f99100;
}

.patternstatusflags__flag-stable {
  background: #38a72f;
}
