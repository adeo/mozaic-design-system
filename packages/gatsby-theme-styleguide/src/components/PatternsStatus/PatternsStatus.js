import React from 'react'
import { useStaticQuery, graphql, Link } from 'gatsby'
import { ControlCross16, NotificationCircleAvailable24 } from '../icons'
import * as styles from './patternstatus.module.css'

const query = graphql`
  query PatternsStatusQuery {
    Foundations: allMdx(
      sort: { frontmatter: { title: ASC } }
      filter: {
        fields: {
          fileName: { name: { eq: "index" } }
          slug: { regex: "/^/Foundations/" }
        }
        frontmatter: { status: {} }
      }
    ) {
      totalCount
      edges {
        node {
          fields {
            slug
          }
          frontmatter {
            title
            order
            status {
              sketch
              scss
              figma
            }
          }
        }
      }
    }
    Components: allMdx(
      sort: { frontmatter: { title: ASC } }
      filter: {
        fields: {
          fileName: { name: { eq: "index" } }
          slug: { regex: "/^/Components/" }
        }
        frontmatter: { status: {} }
      }
    ) {
      totalCount
      edges {
        node {
          fields {
            slug
          }
          frontmatter {
            title
            order
            status {
              sketch
              scss
              figma
            }
          }
        }
      }
    }
  }
`

const PatternsStatus = ({ type }) => {
  const data = useStaticQuery(query)
  const edges = data[type].edges

  const tableModel = edges[0].node.fields.slug.includes('Components')
    ? {
        Patterns: {
          name: null,
        },
        Design: {
          Sketch: null,
          Figma: null,
        },
        Development: {
          SCSS: null,
          Vue: null,
          Freemarker: null,
        },
      }
    : {
        Patterns: {
          name: null,
        },
        Design: {
          Sketch: null,
          Figma: null,
        },
        Development: {
          SCSS: null,
        },
      }

  const categories = Object.keys(tableModel)

  const platforms = categories
    .map((title) => Object.keys(tableModel[title]).map((subtitle) => subtitle))
    .toString()
    .split(',')
    .map((item) => item)

  const platformsTableHeader = platforms.map((subtitle) => (
    <th className={styles.platform} key={'th_' + subtitle}>
      {subtitle === 'name' ? '' : subtitle}
    </th>
  ))

  const discipline = categories.map((title) => {
    const colspan = Object.keys(tableModel[title]).length

    return (
      <th colSpan={colspan} key={'th_' + title}>
        <div className={styles.discipline}>{title}</div>
      </th>
    )
  })

  const tblRows = edges
    .filter((pattern) => !!pattern.node.frontmatter.status)
    .map((pattern, index) => {
      const thisPatternStatus = []

      platforms.forEach((plateform) => {
        const relatedStatus = Object.keys(pattern.node.frontmatter.status).find(
          (status) => status.toLowerCase() === plateform.toLowerCase()
        )

        if (plateform === 'name') {
          thisPatternStatus.push(
            <td key={'value_row_' + plateform + index}>
              <Link
                className={styles.patternName}
                to={pattern.node.fields.slug}
              >
                {pattern.node.frontmatter.title}
              </Link>
            </td>
          )
        } else if (
          relatedStatus &&
          pattern.node.frontmatter.status[relatedStatus]
        ) {
          thisPatternStatus.push(
            <td key={'value_row_' + plateform + index}>
              <NotificationCircleAvailable24 fill="#41a017" />
            </td>
          )
        } else {
          thisPatternStatus.push(
            <td key={'value_row_' + plateform + index}>
              <ControlCross16 fill="currentColor" />
            </td>
          )
        }
      })

      return (
        <tr className={styles.row} key={'value_row_' + index}>
          {thisPatternStatus}
        </tr>
      )
    })

  return (
    <div>
      <table className={styles.statusTable}>
        <thead>
          <tr>{discipline}</tr>
          <tr>{platformsTableHeader}</tr>
        </thead>
        <tbody>{tblRows}</tbody>
      </table>
    </div>
  )
}

export default PatternsStatus
