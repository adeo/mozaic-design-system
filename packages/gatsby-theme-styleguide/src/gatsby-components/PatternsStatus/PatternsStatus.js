import React from 'react'
import styled from 'styled-components'
import { StaticQuery, graphql, Link } from 'gatsby'
import StatusFlag from '../StatusFlag'
import { ControlCross16 } from '@mozaic-ds/icons/react'

export const StatusTable = styled.table`
  font-size: 0.875rem;
  color: #525252;

  td,
  th {
    padding: 0.8rem 0.4rem 1rem;
    line-height: 1.3;
  }

  tr {
    border: 0 !important;
  }

  th {
    border: 0 !important;
  }
`

const Discipline = styled.div`
  border-bottom: solid 1px #554f52;
  font-size: 1rem;
  line-height: 2.5em;
`

const PatternName = styled(Link)`
  display: block;
  font-weight: bold;
  padding-left: 0.5rem;
  width: 110px;
  max-width: 110px;
  color: #525252;

  &:hover {
    text-decoration: underline;
  }
`

const Row = styled.tr`
  border: 0;

  &:nth-child(odd) {
    background-color: #f5f5f5;
  }
`

const Platform = styled.th`
  font-weight: normal;
  color: #999;
  padding-top: 0.6rem !important;
  padding-bottom: 0.4rem !important;
`

const PatternsStatus = ({ data }) => {
  const edges = data.edges

  const tableModel = {
    Patterns: {
      name: null,
    },
    Design: {
      Sketch: null,
      Figma: null,
    },
    Development: {
      SCSS: null,
      Vue: null,
      React: null,
      Angular: null,
    },
  }

  const categories = Object.keys(tableModel)

  const platforms = categories
    .map(title => Object.keys(tableModel[title]).map(subtitle => subtitle))
    .toString()
    .split(',')
    .map(item => item)

  const platformsTableHeader = platforms.map(subtitle => (
    <Platform key={'th_' + subtitle}>
      {subtitle === 'name' ? '' : subtitle}
    </Platform>
  ))

  const discipline = categories.map(title => {
    const colspan = Object.keys(tableModel[title]).length

    return (
      <th colSpan={colspan} key={'th_' + title}>
        <Discipline>{title}</Discipline>
      </th>
    )
  })

  const tblRows = edges
    .filter(pattern => !!pattern.node.frontmatter.status)
    .map((pattern, index) => {
      const thisPatternStatus = []

      platforms.forEach(plateform => {
        const relatedStatus = Object.keys(pattern.node.frontmatter.status).find(
          status => status.toLowerCase() === plateform.toLowerCase()
        )

        if (plateform === 'name') {
          thisPatternStatus.push(
            <td key={'value_row_' + plateform + index}>
              <PatternName to={pattern.node.fields.slug}>
                {pattern.node.frontmatter.title}
              </PatternName>
            </td>
          )
        } else if (relatedStatus) {
          thisPatternStatus.push(
            <td key={'value_row_' + plateform + index}>
              <StatusFlag
                status={pattern.node.frontmatter.status[relatedStatus]}
              />
            </td>
          )
        } else {
          thisPatternStatus.push(
            <td key={'value_row_' + plateform + index}>
              <ControlCross16 fill="currentColor" />
            </td>
          )
        }
      })

      return <Row key={'value_row_' + index}>{thisPatternStatus}</Row>
    })

  return (
    <div>
      <StatusTable>
        <thead>
          <tr>{discipline}</tr>
          <tr>{platformsTableHeader}</tr>
        </thead>
        <tbody>{tblRows}</tbody>
      </StatusTable>
    </div>
  )
}

const query = graphql`
  query PatternsStatusQuery {
    Foundations: allMdx(
      sort: { fields: [frontmatter___title], order: ASC }
      filter: {
        fields: {
          fileName: { name: { eq: "index" } }
          slug: { regex: "/^/Foundations/" }
        }
        frontmatter: { status: {} }
      }
    ) {
      totalCount
      edges {
        node {
          fields {
            slug
          }
          frontmatter {
            title
            order
            status {
              sketch
              scss
            }
          }
        }
      }
    }
    Components: allMdx(
      sort: { fields: [frontmatter___title], order: ASC }
      filter: {
        fields: {
          fileName: { name: { eq: "index" } }
          slug: { regex: "/^/Components/" }
        }
        frontmatter: { status: {} }
      }
    ) {
      totalCount
      edges {
        node {
          fields {
            slug
          }
          frontmatter {
            title
            order
            status {
              sketch
              scss
            }
          }
        }
      }
    }
  }
`

const PatternStatus = () => (
  <StaticQuery
    query={query}
    render={data => (
      <div>
        <h2>Foundations</h2>
        <PatternsStatus data={data.Foundations} />
        <h2>Components</h2>
        <PatternsStatus data={data.Components} />
        <h3>Statuses meaning</h3>
        <p>
          <StatusFlag status="wip" />
          <b>&nbsp;&nbsp;Work In Progress</b> : The pattern should not be used
          because it is very likelly to change in major ways. Generally, wip
          patterns will neither be documented or provide code through this
          website.
        </p>
        <p>
          <StatusFlag status="bêta" />
          <b>&nbsp;&nbsp;Bêta</b> : The pattern is documented and accessible for
          the specified platform but breaking changes may occur before being
          labeled as stable. Basicaly, a bêta version of a pattern is waiting
          for users (you) to test it and confirm us that it fit the teams needs.
          You should avoid using it in production but if you do, be extra
          careful when updating Mozaic.
        </p>
        <p>
          <StatusFlag status="stable" />
          <b>&nbsp;&nbsp;Stable</b> : The pattern is usable in production. The
          Mozaic's team can add features or fix bugs to it as long there is no
          changes required from the final user. Features supported in a stable
          pattern will be supported at least until the next major version (from
          1.5.72 to 2.0.0 for example).
        </p>
      </div>
    )}
  />
)

export default PatternStatus
