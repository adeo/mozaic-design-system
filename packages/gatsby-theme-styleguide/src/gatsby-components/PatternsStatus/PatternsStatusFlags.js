import React from 'react'
import PatternStatusFlagBox from './PatternStatusFlagBox'

import './patternsstatusflags.scss'

/*
  ##### Pattern Status aka SANITY/READINESS/STABILITY Flags #####
*/

const PatternsStatusFlags = props => {
  if (!props || !props.status) {
    return false
  }
  const { status } = props

  const items = Object.entries(status).map((statuses, index) => {
    let [key, value] = statuses
    if (key === 'scss') {
      key = key.toUpperCase()
    }

    return (
      <p key={index} className="patternstatusflags__item">
        {key} <PatternStatusFlagBox value={value} />
      </p>
    )
  })

  if (!items.length) {
    return false
  }

  return <div className="patternstatusflags">{items}</div>
}

export default PatternsStatusFlags
