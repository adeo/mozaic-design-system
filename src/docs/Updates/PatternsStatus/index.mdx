---
title: 'Patterns Status'
order: 1
description: 'The right place to check if the component you want to use is in progress, in bÃªta version, or stable'
---

import PatternsStatus from '@mozaic-ds/gatsby-theme-styleguide/src/gatsby-components/PatternsStatus'

<PatternsStatus />
