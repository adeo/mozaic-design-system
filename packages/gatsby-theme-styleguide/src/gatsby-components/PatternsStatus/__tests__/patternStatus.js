import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { StaticQuery } from 'gatsby'
import PatternsStatus, { StatusTable } from '../PatternsStatus'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      Foundations: {
        totalCount: 2,
        edges: [
          {
            node: {
              fields: {
                slug: '/Foundations/Typography/BodyStyles/',
              },
              frontmatter: {
                title: 'Body styles',
                order: 4,
                status: {
                  sketch: 'bêta',
                  scss: 'bêta',
                },
              },
            },
          },
          {
            node: {
              fields: {
                slug: '/Foundations/BorderRadius/',
              },
              frontmatter: {
                title: 'Border radius',
                order: 8,
                status: {
                  sketch: 'bêta',
                  scss: 'bêta',
                },
              },
            },
          },
        ],
      },
      Components: {
        totalCount: 2,
        edges: [
          {
            node: {
              fields: {
                slug: '/Components/Typography/BodyStyles/',
              },
              frontmatter: {
                title: 'Body styles',
                order: 4,
                status: {
                  sketch: 'bêta',
                  scss: 'bêta',
                },
              },
            },
          },
          {
            node: {
              fields: {
                slug: '/Components/Typography/BodyStyles/',
              },
              frontmatter: {
                title: 'Border radius',
                order: 8,
                status: {
                  sketch: 'bêta',
                  scss: 'bêta',
                },
              },
            },
          },
        ],
      },
    })
  )
})

describe(`Pattern Status Component`, () => {
  it(`render correctly`, () => {
    const tree = renderer.create(<PatternsStatus />)
    const treeJSON = tree.toJSON()
    expect(treeJSON).toMatchSnapshot()
    expect(tree.root.findAllByType(StatusTable).length).toEqual(2)
    expect(tree.root.findAllByType('h2').length).toEqual(2)
  })
})
