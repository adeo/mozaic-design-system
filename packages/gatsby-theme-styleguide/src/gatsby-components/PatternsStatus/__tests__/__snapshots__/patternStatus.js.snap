// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pattern Status Component render correctly 1`] = `
.c5 {
  display: inline-block;
  font-size: 0.875rem;
  line-height: 1;
  padding: 0.2rem 0.5rem 0.3rem;
  background: #fff;
  border-radius: 3px;
  border: solid 1px #e35876;
  color: #82325e;
}

.c6 {
  display: inline-block;
  font-size: 0.875rem;
  line-height: 1;
  padding: 0.2rem 0.5rem 0.3rem;
  background: #fff;
  border-radius: 3px;
  border: solid 1px #e3cb58;
  color: #a58e1b;
}

.c7 {
  display: inline-block;
  font-size: 0.875rem;
  line-height: 1;
  padding: 0.2rem 0.5rem 0.3rem;
  background: #fff;
  border-radius: 3px;
  border: solid 1px #58e3b7;
  color: #328269;
}

.c0 {
  font-size: 0.875rem;
  color: #525252;
}

.c0 td,
.c0 th {
  padding: 0.8rem 0.4rem 1rem;
  line-height: 1.3;
}

.c0 tr {
  border: 0 !important;
}

.c0 th {
  border: 0 !important;
}

.c1 {
  border-bottom: solid 1px #554f52;
  font-size: 1rem;
  line-height: 2.5em;
}

.c4 {
  display: block;
  font-weight: bold;
  padding-left: 0.5rem;
  width: 110px;
  max-width: 110px;
  color: #525252;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  border: 0;
}

.c3:nth-child(odd) {
  background-color: #f5f5f5;
}

.c2 {
  font-weight: normal;
  color: #999;
  padding-top: 0.6rem !important;
  padding-bottom: 0.4rem !important;
}

<div>
  <h2>
    Foundations
  </h2>
  <div>
    <table
      className="c0"
    >
      <thead>
        <tr>
          <th
            colSpan={1}
          >
            <div
              className="c1"
            >
              Patterns
            </div>
          </th>
          <th
            colSpan={2}
          >
            <div
              className="c1"
            >
              Design
            </div>
          </th>
          <th
            colSpan={4}
          >
            <div
              className="c1"
            >
              Development
            </div>
          </th>
        </tr>
        <tr>
          <th
            className="c2"
          >
            
          </th>
          <th
            className="c2"
          >
            Sketch
          </th>
          <th
            className="c2"
          >
            Figma
          </th>
          <th
            className="c2"
          >
            SCSS
          </th>
          <th
            className="c2"
          >
            Vue
          </th>
          <th
            className="c2"
          >
            React
          </th>
          <th
            className="c2"
          >
            Angular
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="c3"
        >
          <td>
            <a
              className="c4"
              href="/Foundations/Typography/BodyStyles/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Body styles
            </a>
          </td>
          <td>
            <span
              className="c5"
            >
              bêta
            </span>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <span
              className="c5"
            >
              bêta
            </span>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
        </tr>
        <tr
          className="c3"
        >
          <td>
            <a
              className="c4"
              href="/Foundations/BorderRadius/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Border radius
            </a>
          </td>
          <td>
            <span
              className="c5"
            >
              bêta
            </span>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <span
              className="c5"
            >
              bêta
            </span>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h2>
    Components
  </h2>
  <div>
    <table
      className="c0"
    >
      <thead>
        <tr>
          <th
            colSpan={1}
          >
            <div
              className="c1"
            >
              Patterns
            </div>
          </th>
          <th
            colSpan={2}
          >
            <div
              className="c1"
            >
              Design
            </div>
          </th>
          <th
            colSpan={4}
          >
            <div
              className="c1"
            >
              Development
            </div>
          </th>
        </tr>
        <tr>
          <th
            className="c2"
          >
            
          </th>
          <th
            className="c2"
          >
            Sketch
          </th>
          <th
            className="c2"
          >
            Figma
          </th>
          <th
            className="c2"
          >
            SCSS
          </th>
          <th
            className="c2"
          >
            Vue
          </th>
          <th
            className="c2"
          >
            React
          </th>
          <th
            className="c2"
          >
            Angular
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="c3"
        >
          <td>
            <a
              className="c4"
              href="/Components/Typography/BodyStyles/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Body styles
            </a>
          </td>
          <td>
            <span
              className="c5"
            >
              bêta
            </span>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <span
              className="c5"
            >
              bêta
            </span>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
        </tr>
        <tr
          className="c3"
        >
          <td>
            <a
              className="c4"
              href="/Components/Typography/BodyStyles/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Border radius
            </a>
          </td>
          <td>
            <span
              className="c5"
            >
              bêta
            </span>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <span
              className="c5"
            >
              bêta
            </span>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
          <td>
            <svg
              fill="currentColor"
              height="1rem"
              viewBox="0 0 16 16"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"
              />
            </svg>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h3>
    Statuses meaning
  </h3>
  <p>
    <span
      className="c6"
    >
      wip
    </span>
    <b>
        Work In Progress
    </b>
     : The pattern should not be used because it is very likelly to change in major ways. Generally, wip patterns will neither be documented or provide code through this website.
  </p>
  <p>
    <span
      className="c5"
    >
      bêta
    </span>
    <b>
        Bêta
    </b>
     : The pattern is documented and accessible for the specified platform but breaking changes may occur before being labeled as stable. Basicaly, a bêta version of a pattern is waiting for users (you) to test it and confirm us that it fit the teams needs. You should avoid using it in production but if you do, be extra careful when updating Mozaic.
  </p>
  <p>
    <span
      className="c7"
    >
      stable
    </span>
    <b>
        Stable
    </b>
     : The pattern is usable in production. The Mozaic's team can add features or fix bugs to it as long there is no changes required from the final user. Features supported in a stable pattern will be supported at least until the next major version (from 1.5.72 to 2.0.0 for example).
  </p>
</div>
`;
